src/Numerical/RootFinding/QuadraticEquation.cpp

```diff
@@ -0,0 +1,24 @@
+
+#include "Numerical/RootFinding/QuadraticEquation.hpp"
+
+#include <gsl/gsl_poly.h>
+
+#include "ErrorHandling/Assert.hpp"
+#include "ErrorHandling/ExpectsAndEnsures.hpp"
+
+double positive_root(const double a, const double b, const double c) {
+  double x0;
+  double x1;
+  gsl_poly_solve_quadratic(a, b, c, &x0, &x1);
+  Ensures(x0 <= 0.0 and x1 >= 0.0);
+  return x1;
```

Comment:
The GSL [documentation](https://www.gnu.org/software/gsl/manual/html_node/Quadratic-Equations.html) indicates that if `gsl_poly_solve_quadratic` finds 0 or 1 real roots, then one or both of `x0`,`x1` will _not_ be modified. If this occurs then `x1` and potentially `x0` remain uninitialized. Spooky.

I would suggest,
* assigning  (`0`, `signaling_nan`, ...?) to both variables at declaration,
* adding an `ASSERT(numrealroots == 2, ...)` as in the function below.